File: benchmarks/pandas/bench_cat_value_counts.py
"""Benchmark: cat_value_counts — pandas Categorical value_counts on 100k-element Series"""
import json
import time
import pandas as pd

ROWS = 100_000
WARMUP = 3
ITERATIONS = 10

cats = ["a", "b", "c", "d", "e"]
s = pd.Categorical([cats[i % len(cats)] for i in range(ROWS)], categories=cats)

for _ in range(WARMUP):
    _ = pd.Series(s).value_counts()

start = time.perf_counter()
for _ in range(ITERATIONS):
    _ = pd.Series(s).value_counts()
total = (time.perf_counter() - start) * 1000

print(json.dumps({"function": "cat_value_counts", "mean_ms": total / ITERATIONS, "iterations": ITERATIONS, "total_ms": total}))

File: benchmarks/tsb/bench_cat_value_counts.ts
/**
 * Benchmark: cat_value_counts — CategoricalAccessor.valueCounts() on 100k-element Series
 */
import { Series } from "../../src/index.js";

const ROWS = 100_000;
const WARMUP = 3;
const ITERATIONS = 10;

const cats = ["a", "b", "c", "d", "e"];
const s = new Series({ data: Array.from({ length: ROWS }, (_, i) => cats[i % cats.length]) });

for (let i = 0; i < WARMUP; i++) {
  s.cat.valueCounts();
}

const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  s.cat.valueCounts();
}
const total = performance.now() - start;

console.log(
  JSON.stringify({
    function: "cat_value_counts",
    mean_ms: total / ITERATIONS,
    iterations: ITERATIONS,
    total_ms: total,
  }),
);
